### PR TITLE
Print client data menu item

### PR DESCRIPTION
**Pull request by RaasAhsan**

### CODE REVIEW

File: src/main/java/com/rsclauncher/menu/PrintClientData.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  @Override
+  public void actionPerformed(ActionEvent e) {
+    try {
+      final JsonFactory factory = new JsonFactory();
+      final JsonGenerator generator = factory.createGenerator(System.out);
+
+      generator.writeStartObject();
+
+      {
```
Comment (forestbelton) [inline src/main/java/com/rsclauncher/menu/PrintClientData.java]: ?

File: src/main/java/com/rsclauncher/menu/PrintClientData.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        writeGameCharacter(generator, client.getLocalPlayer());
+      }
+
+      {
+        generator.writeFieldName("nearbyPlayers");
+        generator.writeStartArray();
+
+        for (int i = 0; i < client.getNearbyPlayers().length; i++) {
+          // TODO: Write null in arrays or not? Otherwise might be real huge
```
Comment (forestbelton) [inline src/main/java/com/rsclauncher/menu/PrintClientData.java]: What I did when printing static arrays was to print `[index] - value` for non-null values. Maybe we could do JSON `{"index": value, ...}`?

File: src/main/java/com/rsclauncher/menu/PrintClientData.java (hunk 1 above)
Comment (RaasAhsan) [reply src/main/java/com/rsclauncher/menu/PrintClientData.java]: just for consistency to clearly separate each object

File: src/main/java/com/rsclauncher/patcher/ClientPatcher.java [hunk 3]
```diff
@@ -83,7 +83,7 @@ public void visitEnd() {
       final MethodVisitor mv = cv.visitMethod(ACC_PUBLIC, "getLocalPlayer", "()Lcom/rsclauncher/api/GameCharacter;", null, null);
       mv.visitCode();
       mv.visitVarInsn(ALOAD, 0);
-      mv.visitFieldInsn(GETFIELD, O_CLASS_NAME, O_GET_LOCAL_PLAYER, "Lcom/rsclauncher/api/GameCharacter;");
+      mv.visitFieldInsn(GETFIELD, O_CLASS_NAME, O_GET_LOCAL_PLAYER, "Lnb;");
```
Comment (forestbelton) [inline src/main/java/com/rsclauncher/patcher/ClientPatcher.java:86]: These code blocks look pretty similar. Is this something we can put in an array and DRY up or no?

File: src/main/java/com/rsclauncher/menu/PrintClientData.java (hunk 1 above)
Comment (forestbelton) [reply src/main/java/com/rsclauncher/menu/PrintClientData.java]: Let's remove em

File: src/main/java/com/rsclauncher/patcher/ClientPatcher.java (hunk 3 above)
Comment (RaasAhsan) [reply src/main/java/com/rsclauncher/patcher/ClientPatcher.java:86]: can be deduped, i've been thinking of the best way to do it, along with maintaining a mapping of obbed -> deobbed names

File: src/main/java/com/rsclauncher/menu/PrintClientData.java (hunk 1 above)
Comment (RaasAhsan) [reply src/main/java/com/rsclauncher/menu/PrintClientData.java]: yes sir

**Pull request merged**